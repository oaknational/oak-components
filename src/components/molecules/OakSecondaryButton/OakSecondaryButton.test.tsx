import React from "react";
import "@testing-library/jest-dom";

import {
  OakSecondaryButton,
  OakSecondaryButtonProps,
} from "./OakSecondaryButton";

import renderWithTheme from "@/test-helpers/renderWithTheme";

const defaultArgs: OakSecondaryButtonProps = {
  iconName: "arrow-right",
};

describe("OakSecondaryButton", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });

  it("renders", () => {
    const { getByTestId } = renderWithTheme(
      <OakSecondaryButton {...defaultArgs} data-testid="test">
        Click
      </OakSecondaryButton>,
    );
    expect(getByTestId("test")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = renderWithTheme(
      <OakSecondaryButton {...defaultArgs}>Click Me</OakSecondaryButton>,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders the chidren", () => {
    const { getByText } = renderWithTheme(
      <OakSecondaryButton {...defaultArgs}>Click</OakSecondaryButton>,
    );
    expect(getByText("Click")).toBeInTheDocument();
  });

  it("calls onClick method", () => {
    const onClick = jest.fn();
    const { getByTestId } = renderWithTheme(
      <OakSecondaryButton {...defaultArgs} data-testid="test" onClick={onClick}>
        Click
      </OakSecondaryButton>,
    );
    getByTestId("test").click();
    expect(onClick).toHaveBeenCalled();
  });
});
