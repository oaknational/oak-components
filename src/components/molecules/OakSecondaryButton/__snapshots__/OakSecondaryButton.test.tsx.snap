// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakSecondaryButton matches snapshot 1`] = `
.c0:where(span) {
  display: block;
}

.c0:where(*) {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c2:where(span) {
  display: block;
}

.c2:where(*) {
  position: absolute;
  top: 0rem;
  width: 100%;
  height: 100%;
  border-radius: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c5:where(span) {
  display: block;
}

.c5:where(*) {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c7:where(span) {
  display: block;
}

.c7:where(*) {
  position: relative;
  width: 1.5rem;
  min-width: 1.5rem;
  height: 1.5rem;
  min-height: 1.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c8 {
  -webkit-filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
  filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
  object-fit: contain;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.5rem;
}

.c9 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
  text-align: left;
}

.c3 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  text-align: left;
  font-family: unset;
  outline: none;
  font-family: --var(google-font),Lexend,sans-serif;
  color: #222222;
  background: #ffffff;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  border: 0.125rem solid;
  border-color: #222222;
  border-radius: 0.25rem;
}

.c3:disabled {
  pointer-events: none;
  cursor: default;
}

.c4 {
  position: relative;
  width: 100%;
  height: 100%;
  display: inline-block;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #222222;
  background: #f2f2f2;
  border-color: #222222;
}

.c4:active {
  background: #ffffff;
  border-color: #222222;
  color: #222222;
}

.c4:disabled {
  background: #e4e4e4;
  border-color: #808080;
  color: #808080;
}

.c1 .grey-shadow:has(+ * + .internal-button:focus-visible) {
  box-shadow: 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c1 .yellow-shadow:has(+ .internal-button:focus-visible) {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%);
}

.c1 .yellow-shadow:has(+ .internal-button:hover),
.c1 .yellow-shadow:has(+ .internal-button:hover:not(:focus-visible,:active)) {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%);
}

.c1 .grey-shadow:has(+ * + .internal-button:hover) {
  box-shadow: none;
}

.c1 .grey-shadow:has(+ * + .internal-button:active) {
  box-shadow: 0.25rem 0.25rem 0 rgba(87,87,87,100%);
}

.c1 .yellow-shadow:has(+ .internal-button:active) {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%);
}

<div
  className="c0 c1"
>
  <div
    className="c2 grey-shadow"
  />
  <div
    className="c2 yellow-shadow"
  />
  <button
    className="c3 c4 internal-button"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="c5 c6"
    >
      <span
        className="c7"
      >
        <img
          alt=""
          className="c8"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1707149070/icons/fv0z57zerrioft52dd9n.svg"
          style={
            {
              "bottom": 0,
              "color": "transparent",
              "height": "100%",
              "left": 0,
              "objectFit": undefined,
              "objectPosition": undefined,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
        />
      </span>
      <span
        className="c9"
      >
        Click Me
      </span>
    </div>
  </button>
</div>
`;
